ui/shared/components/panel/genes/GeneExpression.jsx

```diff
@@ -32,55 +67,85 @@ const xcoord = (d) => {
   return e
 }
 
-const GeneExpression = ({ expression }) => {
-  if (!expression) {
-    return <p>Expression data not available for this gene.</p>
+class GeneExpression extends React.PureComponent {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      popupOpen: false,
+      hoverData: { x: 0, y: 0, sampleCount: 0 },
+      hoverPostiton: { pageX: 0, pageY: 0 },
+    }
+  }
+
+  mouseOverHandler = (d, e) => {
+    this.setState({
+      popupOpen: true,
+      hoverData: d,
+      hoverPostiton: e,
+    })
   }
 
-  const tissues = Object.keys(expression).sort()
-  const data = tissues.reduce((acc, tissue) =>
-    acc.concat(expression[tissue].filter(d => d > 0).map((d) => { return {
-      x: xcoord(d),
-      y: tissueName(tissue),
-      type: tissue,
-    } }),
-    ), [])
-  const config = tissues.map((tissue, i) => { return { type: tissue, color: COLORS[i % COLORS.length] } })
-
-  const style = {
-    // Make dots more transparent
-    '.dot': { opacity: 0.12 },
-    // Hide unwanted axis lines
-    '.axis path': { display: 'none' },
-    '.x .tick-circle': { display: 'none' },
-    // Move x axis to top and format text
-    '.x.axis, .x.axis .label': { transform: 'translate(0, -20px)', opacity: 0.5, 'font-size': '11px', 'font-weight': '600', fill: 'black' },
-    '.x.axis .label': { transform: 'translate(-790px, 6px)', opacity: 1 },
-    // Format y axis
-    '.y.axis': { 'font-size': '14px', 'font-weight': '600' },
-    '.y .tick-circle': { r: '4px', opacity: 0.7 },
-    // Make y axis tick circles match the dot color
-    ...config.reduce((acc, cfg, i) => {
-      return { [`.y g:nth-child(${i + 2}) .tick-circle`]: { fill: cfg.color }, ...acc }
-    }, {}),
+  mouseOutHandler = () => {
+    // this.setState({ popupOpen: false })
   }
 
-  return (
-    <Segment padded>
-      <ScatterplotChart
-        axes
-        yType="text"
-        width={1200}
-        height={750}
-        data={data}
-        config={config}
-        xDomainRange={RANGE}
-        axisLabels={X_LABELS}
-        style={style}
-        margin={MARGIN}
-      />
-    </Segment>
-  )
+  render() {
+    const { expression } = this.props
+    if (!expression) {
+      return <p>Expression data not available for this gene.</p>
+    }
+
+    const tissues = Object.keys(expression).sort()
+    const tissueExpression = tissues.reduce((acc, tissue) =>
+      ({ ...acc, [tissue]: expression[tissue].filter(d => d > 0) }), {},
```

Comment:
note: we'll want to move this filter to when preprocessing/loading the expression matrix into elasticsearch